### ISSUE TITLE
UI Shell switcher loses focus but remains expanded

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#10349]
### Package

carbon-components

### Browser

Chrome, Safari

### Operating System

MacOS

### Package version

version on carbondesignsystem.com

### Automated testing tool and ruleset

n/a

### Assistive technology

keyboard

### Description

When a user opens the content switcher menu (aka Site) ![image.png](https://zenhub.ibm.com/images/5cab73f442b8dc5ede7b84cc/02ed79fd-ba51-473b-9105-2aa777876f3e) the user must tab through the menu items (okay, but not optimal; should be arrow). However, since the Tab ring is not restricted to the menu, it is possible to leave the menu and navigate to the main content, leaving the menu obscuring content. (Shown in the below screen shot, where the focus is clearly on the Develop tile, which is obscured by the menu.) 

Fortunately the user can dismiss the menu with Esc at any point, or I would rate this higher priority

One of two things should occur:

1. tab is restricted to the menu (just wraps around to top item) until dismissed with Esc (or Enter on X)
1. the menu is dismissed automatically when the focus leaves the last menu item

As mentioned earlier, I would advocate that this menu have arrow key interaction, but that is not as critical as having this menu not persist after it loses focus.

![image.png](https://zenhub.ibm.com/images/5cab73f442b8dc5ede7b84cc/71a91399-cf0d-494b-bc7f-c5471de1178e)

Potentially linked with https://github.com/carbon-design-system/carbon/issues/9519

### WCAG 2.1 Violation

It will likely violate WCAG 2.2 Focus Appearance, since it is likely to obscure an item with focus

### CodeSandbox example

can be replicated on carbondesignsystem.com

### Steps to reproduce

1. tab to the content switcher
2. press Enter to open
3. press tab until the focus has exited the content switcher

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by mbgower** [comment]
@guidari I assume it's intentional that the switcher is open in the React storybook -- as some way to draw the user to how it looks open. But that when implemented, this would not be opened until activated. Correct?

**Comment by guidari** [comment]
> @guidari I assume it's intentional that the switcher is open in the React storybook -- as some way to draw the user to how it looks open. But that when implemented, this would not be opened until activated. Correct?

Exactly, the user can control the switcher state. It's only open to draw the user attention to it

**Comment by mbgower** [comment]
Sorry to keep commenting on a closed issue. Let me know if this should go somewhere else!

The primary feedback I got from the accessibility team when I invited them to explore it was that it was definitely improved overall. 

One concern was that it seemed odd to have the arrow key navigation only happening once the user had tabbed into the first subcomponent.

Some thought the first item should take focus on open (so the arrowing would work immediately). This seemed to work relatively well when thought through:
1. the user activates the switcher
2. focus moves to first item

- arrow navigation is, as it is now, restricted to the child items; pressing Esc puts focus back on switcher
- tab key is almost exactly the same as now, but focus begins on first item; the only lost functionality is inability to toggle switcher off and on; user can still collapse by tabbing out (to next item in DOM) or shift+tabbing out (to switcher) or Escaping (to switcher)

Another suggestion was to leave as is but have the Down Arrow from the switcher move the user into the first child item, but I think this is a less elegant solution and conforms less to the behaviour of most interactions. It also creates the oddity where the user could arrow down into the children but not arrow up into the switcher -- unless you then made the arrow cycle include the switcher button too.

Happy to discuss this further, and thanks for your other meeting on the UI Shell overall.

**Comment by guidari** [comment]
Don't worry! 
We can open a new issue and tackle those extra changes and discuss what would be the best way to approach that on the code.

Thanks for accepting the invite in the next meeting!
